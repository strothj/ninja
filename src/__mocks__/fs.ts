import fs from "fs";
const mockFs: IMockFileSystem = jest.genMockFromModule("fs");

let readFileSyncFilename: string | null = null;
let readFileSyncFileContents: string | null = null;
let writeFileSyncFilename: string | null = null;
let writeFileSyncFileContents: string | null = null;

function mockReadFileSync(path: string, options: string): string {
  if (options !== "utf8") throw new Error('Expected options to be "utf8".');

  if (readFileSyncFilename === null || readFileSyncFileContents === null) {
    throw new Error("readFileSync mock not set");
  }

  if (path !== readFileSyncFilename) {
    throw new Error(
      `Filename did not match expected: ${readFileSyncFilename}.`,
    );
  }

  const fileContents = readFileSyncFileContents;
  readFileSyncFilename = null;
  readFileSyncFileContents = null;
  return fileContents;
}

function setMockReadFileSync(filename: string, fileContents: string) {
  readFileSyncFilename = filename;
  readFileSyncFileContents = fileContents;
}

function mockWriteFileSync(path: string, data: string, options: string) {
  if (options !== "utf8") {
    throw new Error("Expected encoding to be set to utf8");
  }

  writeFileSyncFilename = path;
  writeFileSyncFileContents = data;
}

function getMockWriteFileSync(): { filename: string; fileContents: string } {
  if (writeFileSyncFilename === null || writeFileSyncFileContents === null) {
    throw new Error("writeFileSync was not called");
  }

  const filename = writeFileSyncFilename;
  const fileContents = writeFileSyncFileContents;
  writeFileSyncFilename = null;
  writeFileSyncFileContents = null;
  return { filename, fileContents };
}

mockFs.readFileSync = mockReadFileSync as any;
mockFs.__setReadFileSync = setMockReadFileSync;
mockFs.writeFileSync = mockWriteFileSync as any;
mockFs.__getWriteFileSync = getMockWriteFileSync;

type FSModule = typeof fs;

export interface IMockFileSystem extends FSModule {
  __setReadFileSync: typeof setMockReadFileSync;
  __getWriteFileSync: typeof getMockWriteFileSync;
}

export default mockFs;
