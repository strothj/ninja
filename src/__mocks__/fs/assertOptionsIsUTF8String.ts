import assert from "assert";

export const assertOptionsIsUTF8String = (options: any) =>
  assert.strictEqual(
    options,
    "utf8",
    `Expected options to be "utf8", got "${JSON.stringify(options, null, 2)}".`,
  );
