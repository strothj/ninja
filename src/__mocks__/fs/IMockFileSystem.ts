import fs from "fs";

type IFileSystemModuleType = typeof fs;

type FileSystemModuleOverrides = "readFileSync" | "writeFileSync";

export interface IMockFileSystem
  extends Omit<IFileSystemModuleType, FileSystemModuleOverrides> {
  readFileSync: (path: string, options: string) => string;
  readFileSync__set: (filename: string, fileContents: string) => void;
  writeFileSync: (path: string, data: string, options: string) => void;
  writeFileSync__get: () => { filename: string; fileContents: string };
}

export type IReadFileSync = IMockFileSystem["readFileSync"];
export type IReadFileSyncSet = IMockFileSystem["readFileSync__set"];

export type IWriteFileSync = IMockFileSystem["writeFileSync"];
export type IWriteFileSyncGet = IMockFileSystem["writeFileSync__get"];
