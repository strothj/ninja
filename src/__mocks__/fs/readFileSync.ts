import assert from "assert";
import { assertOptionsIsUTF8String } from "./assertOptionsIsUTF8String";
import { IReadFileSync, IReadFileSyncSet } from "./IMockFileSystem";

let readFileSync__filename: string | null = null;
let readFileSync__fileContents: string | null = null;

export const readFileSync: IReadFileSync = (path, options) => {
  assertOptionsIsUTF8String(options);

  assert.ok(
    readFileSync__filename && readFileSync__fileContents,
    "readFileSync mock not set.",
  );
  assert.strictEqual(
    path,
    readFileSync__filename,
    `"path" did not match expected: "${readFileSync__filename}", got: "${path}".`,
  );

  const fileContents = readFileSync__fileContents as string;
  readFileSync__filename = null;
  readFileSync__fileContents = null;
  return fileContents;
};

export const readFileSync__set: IReadFileSyncSet = (filename, fileContents) => {
  readFileSync__filename = filename;
  readFileSync__fileContents = fileContents;
};
