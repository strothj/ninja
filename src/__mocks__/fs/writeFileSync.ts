import assert from "assert";
import { assertOptionsIsUTF8String } from "./assertOptionsIsUTF8String";
import { IWriteFileSync, IWriteFileSyncGet } from "./IMockFileSystem";

let writeFileSync__filename: string | null = null;
let writeFileSync__fileContents: string | null = null;

export const writeFileSync: IWriteFileSync = (path, data, options) => {
  assertOptionsIsUTF8String(options);

  writeFileSync__filename = path;
  writeFileSync__fileContents = data;
};

export const writeFileSync__get: IWriteFileSyncGet = () => {
  assert.ok(writeFileSync__filename, "writeFileSync was not called.");

  const response = {
    fileContents: writeFileSync__fileContents as string,
    filename: writeFileSync__filename as string,
  };

  writeFileSync__filename = null;
  writeFileSync__fileContents = null;

  return response;
};
