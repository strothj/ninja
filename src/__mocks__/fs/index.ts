import { IMockFileSystem } from "./IMockFileSystem";
import { readFileSync, readFileSync__set } from "./readFileSync";
import { writeFileSync, writeFileSync__get } from "./writeFileSync";

const mockFs: IMockFileSystem = jest.genMockFromModule("fs");
mockFs.readFileSync = readFileSync;
mockFs.readFileSync__set = readFileSync__set;
mockFs.writeFileSync = writeFileSync;
mockFs.writeFileSync__get = writeFileSync__get;

export default mockFs;
