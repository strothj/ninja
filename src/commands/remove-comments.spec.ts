import tsconfigWithComments from "./__fixtures__/tsconfig-with-comments";
import removeComments from "./remove-comments";

jest.mock("fs");

beforeEach(() => {
  setReadFileSyncMock();
});

it("removes comments", () => {
  removeComments("tsconfig.json");
  const result = getWriteFileSyncMock();
  expect(result.filename).toEqual("tsconfig.json");
  expect(result.fileContents).toMatchSnapshot();
});

function setReadFileSyncMock() {
  require("fs").default.readFileSync__set(
    "tsconfig.json",
    tsconfigWithComments,
  );
}

function getWriteFileSyncMock(): { filename: string; fileContents: string } {
  return require("fs").default.writeFileSync__get();
}
