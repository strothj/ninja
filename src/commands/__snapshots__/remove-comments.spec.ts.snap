// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removes comments 1`] = `
"{
  \\"compilerOptions\\": {
    \\"target\\": \\"es5\\",                          
    \\"module\\": \\"commonjs\\",                     
    \\"strict\\": true,                           
    \\"esModuleInterop\\": true                   
  }
}
"
`;
